stock_picking_auto_create_package/models/stock_picking.py

```diff
@@ -0,0 +1,113 @@
+# Copyright 2023 Jacques-Etienne Baudoux (BCIM) <je@bcim.be>
+# Copyright 2023 Raumschmiede Gmbh
+# Copyright 2023 Michael Tietz (MT Software) <mtietz@mt-software.de>
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+from odoo import api, models
+from odoo.tools import float_compare, float_round
+
+
+class StockPicking(models.Model):
+    _inherit = "stock.picking"
+
+    def _auto_create_delivery_package(self) -> None:
+        self.ensure_one()
+        if self.picking_type_id.automatic_package_creation_mode == "single":
+            self._auto_create_delivery_package_single()
+        elif self.picking_type_id.automatic_package_creation_mode == "packaging":
+            self._auto_create_delivery_package_per_smallest_packaging()
+
+    @api.model
+    def _auto_create_delivery_package_filter(self, move_lines):
+        """
+        Filter move lines that can be put in pack
+        """
+        return move_lines.filtered(
+            lambda ml: not ml.result_package_id
+            and ml.state not in ("cancel", "done")
+            and ml.quantity
```

Comment:
When you have multiple move lines for a move, only some can be picked. Non picked move lines will be deleted by call to action_done occuring afterwards.
When the move is not picked, all move lines will be considered as picked on action_done
```suggestion
            and ml.quantity
            and (not ml.move_id.picked or ml.picked)
```